### PR TITLE
Add checks when deleting client

### LINKED ISSUE #141: Deleting a client does not delete the associated Session and Schedule

### PR DESCRIPTION
**Pull request by dhafinrazaq**
Note: @kelvinvin (as the delete client implementer) might want to add tests/update the documentation

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java [hunk 1]
```diff
@@ -26,10 +26,33 @@
 
     public static final String MESSAGE_DELETE_CLIENT_SUCCESS = "Deleted Client: %1$s";
 
+    public static final String MESSAGE_FORCE_DELETE_CLIENT_USAGE = COMMAND_WORD
+            + ": Cannot delete the Client identified by the index number because there are schedules tied to it.\n"
+            + "To force delete, pass in f/ true as an option. BEWARE, YOU WILL LOSE ALL RELATED SCHEDULES.\n"
+            + "Parameters: INDEX (must be a positive integer) f/ true\n"
+            + "Example: " + COMMAND_WORD + " 1 f/ true";
```
Comment (kelvinvin) [inline src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java]: It doesn't actually check for true right? Both this and delete sessions command will execute the command with just the force prefix

Comment (dhafinrazaq) [reply src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java]: yes, it works even without the true. not sure why @benclmnt also have the true in the cmd usage message

File: src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java [hunk 2]
```diff
@@ -69,14 +69,11 @@ public CommandResult execute(Model model) throws CommandException {
         // Do not delete Client unless user use the force flag
         if (model.hasAnyScheduleAssociatedWithClient(clientToDelete) && !isForced) {
             return new CommandResult(MESSAGE_FORCE_DELETE_CLIENT_USAGE);
+        } else {
```
Comment (benclmnt) [inline src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java]: I think it is better to not have `else` after a return statement. It is following [Make the happy path prominent](https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent)

File: src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java (hunk 1 above)
Comment (benclmnt) [reply src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java]: agreed. based on our discussion in telegram, the docs should document this.

File: src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java (hunk 2 above)
Comment (dhafinrazaq) [reply src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java]: oh @kelvinvin 's suggestion is to have the else to make it clearer
so make it like the prev commit?

**Review by benclmnt — APPROVED** [review]

**Pull request merged**